internal/kubernetes/drainer.go

```diff
@@ -163,32 +174,36 @@ func (d *APICordonDrainer) Cordon(n *core.Node) error {
 
 // Drain the supplied node. Evicts the node of all but mirror and DaemonSet pods.
 func (d *APICordonDrainer) Drain(n *core.Node) error {
-	pods, err := d.getPods(n.GetName())
-	if err != nil {
-		return errors.Wrapf(err, "cannot get pods for node %s", n.GetName())
-	}
-
-	abort := make(chan struct{})
-	errs := make(chan error, 1)
-	for _, pod := range pods {
-		go d.evict(pod, abort, errs)
-	}
-	// This will _eventually_ abort evictions. Evictions may spend up to
-	// d.deleteTimeout() in d.awaitDeletion(), or 5 seconds in backoff before
-	// noticing they've been aborted.
-	defer close(abort)
+	if d.withDrain {
```

Comment:
Instead of adding the guard condition and wrapping the body of the function, the typical golang convention is to e.g.

```
if d.withDrain {
    return nil
}
// The original body of the function.
```
It also leads to a clearer diff.

Reply:
Thanks for the feedback, I've implemented this